webapi/src/test/scala/org/knora/webapi/slice/admin/domain/model/UserSpec.scala

```diff
@@ -0,0 +1,161 @@
+/*
+ * Copyright © 2021 - 2024 Swiss National Data and Service Center for the Humanities and/or DaSCH Service Platform contributors.
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package org.knora.webapi.slice.admin.domain.model
+
+import zio.test.*
+
+import dsp.valueobjects.IriErrorMessages
+
+object UserSpec extends ZIOSpecDefault {
+  private val validUserIri            = "http://rdfh.ch/users/jDEEitJESRi3pDaDjjQ1WQ"
+  private val userIriWithUUIDVersion3 = "http://rdfh.ch/users/cCmdcpn2MO211YYOplR1hQ"
+  private val invalidIri              = "Invalid IRI"
+  private val validPassword           = "pass-word"
+  private val validGivenName          = "John"
+  private val validFamilyName         = "Rambo"
+  private val pwStrength              = PasswordStrength.unsafeMake(12)
+
+  private val userSuite = suite("User")()
+
+  private val usernameSuite = suite("Username")(
+    test("Username must not be empty") {
+      assertTrue(Username.from("") == Left(UserErrorMessages.UsernameMissing))
+    },
+    test("Username may contain alphanumeric characters, underscore and dot") {
+      assertTrue(Username.from("a_2.3").isRight)
+    },
+    test("Username has to be at least 4 characters long") {
+      assertTrue(Username.from("abc") == Left(UserErrorMessages.UsernameInvalid))
+    },
+    test("Username has to be at most 50 characters long") {
+      assertTrue(
+        Username
+          .from("123456789012345678901234567890123456789012345678901")
+          .left
+          .exists(_.equals(UserErrorMessages.UsernameInvalid))
+      )
+    },
+    test("Username must not contain other characters") {
+      assertTrue(
```

Comment:
how about using check?

```
val invalid = List("a_2.3!", …)
check ( Gen.fromIterable(invalid)){ i =>
   assertTrue(Username.from(i) == Left("Username is invalid.")
}
```

The advantage is that this way the test will verify all of the values in `invalid` and it would not fail for the first and not check the others.